### ISSUE TITLE
Improvements to the Limited Staff role experience

### ISSUE THREAD

**Issue by jmakowski1123** [openedx/platform-roadmap#361]
The Limited Staff role needs some improvements to the user experience.

Currently, when a user with the Limited Staff role lands on a Studio Page in the new course-authoring MFE, they are blocked by a 403 error message, which contains json fields and is difficult to interpret. Minimally, the messaging could be improved to explain that the user is not granted access to Studio. 

An enhancement can be made in the Studio experience such that users with this role do not see the "View this course in Studio" button. This will greatly reduce the chance of users trying to access parts of the platform that they don't have access to.

**Comment by jmakowski1123** [comment]
@arbrandes @brian-smith-tcril Could you please add any technical requirements you'd like to see here?

**Comment by brian-smith-tcril** [comment]
In order to be able to fully test the Limited Staff role UX, I'd want to know:
* What pages are not accessible by Limited Staff?
  * If a Limited Staff user navigates directly to those pages (by entering a URL), what should they see?
    * Should they see the same thing on every page they cannot access?
    * Should they see the same thing as other roles that cannot access those pages?
* What UI elements on pages Limited Staff *can* access exist that navigate to pages Limited Staff *cannot* access?
  * Are they currently visible by limited staff?
  * Should they be hidden/disabled?

Answers to the previous questions would help inform my thoughts on this from a technical implementation perspective:
* If Limited Staff should see the same thing on every page they cannot access, and they should see the same thing as other roles that cannot access those pages, can we make a generic "you lack permission to see this page" page?
* If all UI elements that navigate to pages Limited Staff cannot access should be hidden/disabled, and the same is true for any other role that cannot access a page, can we write generic "hide/disable based on role" logic to wrap navigation elements in?

**Comment by jmakowski1123** [comment]
Limited Staff should not have access to any of the Studio pages in the course-authoring MFE.
In the LMS, Limited Staff have the same permissions as Course Staff.
Limited Staff should not see the button to "view this course in Studio" from the LMS. This will reduce chances for Limited Staff to land on Studio pages. 
If Limited Staff land on a Studio page via a url, they should see a message indicating they do not have access to the page. This message can be the same for any Studio url. 

@0x29a as the contributing author for this role, is there anything you'd add here? https://github.com/openedx/edx-platform/pull/32570

**Comment by itsjeyd** [comment]
@jmakowski1123 It looks like the following PRs are providing a partial implementation of this proposal:

* https://github.com/openedx/frontend-app-learning/pull/1436
* https://github.com/openedx/edx-platform/pull/35313

Could you please have a look at those and let us know what the next steps are? 

CC @mphilbrick211

**Comment by itsjeyd** [comment]
@jmakowski1123 Could you please have a look at the comment above and let us know how to proceed?

(Happy to ping someone else from the product working group if you'd prefer. Let me know.)

**Comment by itsjeyd** [comment]
@ali-hugo That's helpful input, thank you!

@0x29a Can you confirm that https://github.com/openedx/frontend-app-learning/pull/1436 and https://github.com/openedx/edx-platform/pull/35313 address this need:

> An enhancement can be made in the Studio experience such that users with this role do not see the "View this course in Studio" button. This will greatly reduce the chance of users trying to access parts of the platform that they don't have access to.

... and nothing else?

**Comment by 0x29a** [comment]
@itsjeyd, can confirm. These two PRs just remove the button in question for Limited Staff and do nothing else.

**Comment by ali-hugo** [comment]
@0x29a @itsjeyd It was just confirmed in the Core Product Working Group meeting that https://github.com/openedx/frontend-app-learning/pull/1436 and https://github.com/openedx/edx-platform/pull/35313 can be merged! 🚀 

We'll keep the current ticket (#361) open to work on improving the user experience when someone visits a Studio page to which they don't have access. We need a more user-friendly solution than 403 error that is currently shown. @0x29a, could you please send a screenshot of the 403 message so I can take a look?

**Comment by ali-hugo** [comment]
> Regarding the 403 message, do we have an internal ticket for working on that?

@itsjeyd Not yet. I am out of hours for this month, so will only be able to work on this next month. I will create an internal ticket then.

**Comment by ali-hugo** [comment]
@0x29a Thank you!

**Comment by itsjeyd** [comment]
@ali-hugo 

>> Regarding the 403 message, do we have an internal ticket for working on that?
>
> Not yet. I am out of hours for this month, so will only be able to work on this next month. I will create an internal ticket then.

Sounds good.

@0x29a If you end up spending more time helping out here, please log that time on the ticket that Ali's going to create. Since the 403 message is out of scope for the original ticket, we can't use that ticket to log time spent on the 403 message.

**Comment by ali-hugo** [comment]
@jmakowski1123 @0x29a I've been thinking about how to improve the 403 messaging when someone with the Limited Staff role visits a Studio URL. Please let me know what you think of the two options below:

**Option 1:**
![Studio 403 - Option 1](https://github.com/user-attachments/assets/73c9d64a-5651-441d-a3f4-674f1981ee87)

**Option 2:**
![Studio 403 - Option 2](https://github.com/user-attachments/assets/6a944b4d-8c31-439f-9d91-08bfe0ca19c3)

You'll notice that I've included a "back to safety" link to the LMS (since the Limited Staff user doesn't have access to any Studio pages). For other types of users, I imagine it would be more helpful to link them to the Studio Home. Is it too complicated to show different 403's for different user types?

Thanks for your help!

**Comment by sarina** [comment]
#2 is better from my opinion, it's more standard a message.

**Comment by ali-hugo** [comment]
@sarina Thanks for the feedback!

What do you think about displaying different messages for users with different permissions?:

> You'll notice that I've included a "back to safety" link to the LMS (since the Limited Staff user doesn't have access to any Studio pages). For other types of users, I imagine it would be more helpful to link them to the Studio Home. Is it too complicated to show different 403's for different user types?

**Comment by sarina** [comment]
I think it may be too complicated to display different error messages, but that's a technical detail that is worth asking the implementation team because it's a good idea.

**Comment by ali-hugo** [comment]
@0x29a We've decided to go with the following for the 403 message:

![381878200-6a944b4d-8c31-439f-9d91-08bfe0ca19c3](https://github.com/user-attachments/assets/90f4b299-26fc-4274-a05f-89e7d13820f3)


Here's the text for easy copy pasting: 

**Access Restricted**
It looks like you’re trying to access a page you don’t have permission to view. Contact your admin if you think this is a mistake, or head back to the LMS.

❓  **QUESTION** ❓  
Do you think it would be feasible to show different error messages for different user types? For example, Limited Staff users, who don’t have access to any Studio pages, could see a message guiding them back to the LMS. Meanwhile, users who _do_ have Studio access could be directed to the Studio Home page. What do you think?

**Comment by sarina** [comment]
I agree with one message!

**Comment by ali-hugo** [comment]
@0x29a That makes sense; let's go with a single message shown to all users:

> Access Restricted
It looks like you’re trying to access a page you don’t have permission to view. Contact your admin if you think this is a mistake, or head back to the LMS.

"LMS" should be a link.

Thanks for your work on this!

**Comment by farhaanbukhsh** [comment]
@ali-hugo 

> @farhaanbukhsh Thanks for offering to add the new 403 message!

https://github.com/openedx/frontend-app-authoring/pull/1569

The PR above takes a stab on the implementation and is up for review.

**Comment by farhaanbukhsh** [comment]
@ali-hugo  this is merged now 😄 🎉

**Comment by ali-hugo** [comment]
Closing this ticket as the following PR's cover the requirements listed in the description:

- https://github.com/openedx/frontend-app-learning/pull/1436
- https://github.com/openedx/edx-platform/pull/35313
- https://github.com/openedx/frontend-app-authoring/pull/1569

CC: @jmakowski1123
